### PR TITLE
Add x86_64 intrinsics guard

### PR DESCRIPTION
**Pull request by dmm-fb**
Test Plan: Run tests

Differential Revision: D38964321

### CODE REVIEW

File: c10/util/llvmMathExtras.h [hunk 1]
```diff
@@ -45,7 +45,9 @@
 #ifdef _MSC_VER
 // Declare these intrinsics manually rather including intrin.h. It's very
 // expensive, and MathExtras.h is popular.
+#if defined(__x86_64__) || defined(__i386__) || defined(_MSC_VER)
```
Comment (albanD) [inline c10/util/llvmMathExtras.h:48]: Not needed here haha

**Review by albanD — COMMENTED** [review]
Looks good to me. 
cc @kimishpatel for the quantization sse code

File: c10/util/llvmMathExtras.h (hunk 1 above)
Comment (dmm-fb) [reply c10/util/llvmMathExtras.h:48]: I'll fix it. Thanks

File: aten/src/ATen/native/quantized/cpu/qnnpack/src/q8avgpool/up8x9-sse2.c [hunk 2]
```diff
@@ -8,7 +8,9 @@
 
 #include <assert.h>
 
+#if defined(__x86_64__) || defined(__i386__) || defined(_MSC_VER)
```
Comment (kimishpatel) [inline aten/src/ATen/native/quantized/cpu/qnnpack/src/q8avgpool/up8x9-sse2.c:11]: Is _MSC_VER necessary?

**Review by kimishpatel — CHANGES_REQUESTED** [review]
Looks good for the most part. Left one comment. How did you validate this?

cc: @digantdesai

File: aten/src/ATen/native/quantized/cpu/qnnpack/src/q8avgpool/up8x9-sse2.c (hunk 2 above)
Comment (dmm-fb) [reply aten/src/ATen/native/quantized/cpu/qnnpack/src/q8avgpool/up8x9-sse2.c:11]: I think we can compile the code under Windows too - I don't think the __x86_64__ is defined there

**Comment by ajtulloch** [comment]
@kimishpatel @digantdesai are your concerns addressed? Please accept if so.

**Review by ajtulloch — APPROVED** [review]
Looks reasonable.

**Review by malfet — APPROVED** [review]
Sure, why not, though, do we really need to check for `defined(__i386__)`?
And wouldn't it be better to define  `C10_X86` in `c10/macros/Macros.h` and use it throughout the code?

And by the way, a way of testing it would be to enable QNNPACK compilation for M1 target

File: aten/src/ATen/native/quantized/cpu/qnnpack/src/q8avgpool/up8x9-sse2.c (hunk 2 above)
Comment (kimishpatel) [reply aten/src/ATen/native/quantized/cpu/qnnpack/src/q8avgpool/up8x9-sse2.c:11]: My worry was compiling windows on arm. So something like `defined(_MSC_VER) && defined(_M_IX86)`? I am not sure if thats the right macro though.

**Review by kimishpatel — APPROVED** [review]
Left a comment but generally ok to land.

**Pull request closed without merging**